pilot/pkg/model/gateway.go

```diff
@@ -45,24 +56,129 @@ func MergeGateways(gateways ...Config) *MergedGateway {
 		log.Debugf("MergeGateways: merging gateway %q into %v:\n%v", name, names, gateway)
 		for _, s := range gateway.Servers {
 			log.Debugf("MergeGateways: gateway %q processing server %v", name, s.Hosts)
-			if ss, ok := servers[s.Port.Number]; ok {
-				// TODO: remove this check when Envoy supports filter chain matching so we can expose multiple protocols on the same physical port
-				// ss must have at least one element because the key exists in the map, otherwise we'd be in the else case below.
-				if ss[0].Port.Protocol != s.Port.Protocol {
-					log.Debugf("skipping server: attempting to merge servers for gateway %q into %v but servers have different protocols: want %v have %v",
-						spec.Name, names, ss[0].Port.Protocol, s.Port.Protocol)
-					continue
+			protocol := ParseProtocol(s.Port.Protocol)
+			if gatewayPorts[s.Port.Number] {
+				// We have two servers on the same port. Should we merge?
+				// 1. Yes if both servers are plain text and HTTP
+				// 2. Yes if both servers are using TLS
+				//    if using HTTPS ensure that port name is distinct so that we can setup separate RDS
```

Comment:
Yes if both servers are using HTTPS, and we're generating the RDS for the default cert.

And I suspect it'll also be yes for multi-SAN SNI - but ok if we're not handling it now ( it should still be included in the comments, for people who read the code later and are not familiar with the nuances of SAN/SNI)

Reply:
How do I figure out if the gateway.Server is handling default cert or multi-SAN SNI ?